src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/PrefetchGetRecordsCache.java

```diff
@@ -69,47 +72,55 @@ public GetRecordsResult getNextResult() {
 
     @Override
     public void shutdown() {
-        executorService.shutdown();
+        executorService.shutdownNow();
     }
-    
+
     private class DefaultGetRecordsCacheDaemon implements Runnable {
         @Override
         public void run() {
             while (true) {
+                if (Thread.interrupted()) {
+                    log.warn("Prefetch thread was interrupted.");
+                    break;
+                }
                 if (prefetchCounters.byteSize < maxByteSize && prefetchCounters.size < maxRecordsCount) {
                     try {
                         GetRecordsResult getRecordsResult = getRecordsRetrievalStrategy.getRecords(maxRecordsPerCall);
-                        getRecordsResultQueue.put(getRecordsResult);
-                        prefetchCounters.added(getRecordsResult);
+                        ProcessRecordsInput processRecordsInput = new ProcessRecordsInput()
+                                .withRecords(getRecordsResult.getRecords())
+                                .withMillisBehindLatest(getRecordsResult.getMillisBehindLatest())
+                                .withCacheEntryTime(Instant.now());
+                        getRecordsResultQueue.put(processRecordsInput);
+                        prefetchCounters.added(processRecordsInput);
                     } catch (InterruptedException e) {
                         log.error("Interrupted while adding records to the cache", e);
```

Comment:
This isn't an error.  This probably should be info, and indicate that a shutdown was likely requested.